File: ios/Model/Message/Incoming/AccountUsernameMessage.swift
//
// Copyright © 2023 Nevis Security AG. All rights reserved.
//

import Foundation

class AccountUsernameMessage: ChannelInMessage {
	// MARK: Properties

	let username: String

	// MARK: Decodable

	enum CodingKeys: String, CodingKey {
		case username
	}

	required init(from decoder: Decoder) throws {
		let container = try decoder.container(keyedBy: CodingKeys.self)
		self.username = try container.decode(String.self, forKey: .username)
		try super.init(from: decoder)
	}
}

File: android/src/main/java/ch/nevis/mobile/authentication/sdk/react/model/message/incoming/AccountUsernameMessage.kt
/*
 * Copyright © 2023 Nevis Security AG. All rights reserved.
 */

package ch.nevis.mobile.authentication.sdk.react.model.message.incoming

import kotlinx.serialization.Serializable

@Serializable
data class AccountUsernameMessage(
    override val operationId: String,
    val username: String
) : ChannelInMessage()
